addons/website_rating_project/models/project.py

```diff
@@ -23,3 +23,108 @@ def action_view_all_rating(self):
                 'url': "/project/rating/%s" % (self.id,)
             }
         return super(Project, self).action_view_all_rating()
+
+    @api.multi
+    def _compute_website_url(self):
+        super(Project, self)._compute_website_url()
+        for project in self:
+            project.website_url = "/project/rating/%s" % project.id
+
+
+class ProjectRating(models.AbstractModel):
+    _name = 'project.rating'
+
+    def _get_partner_rating(self, table, model, project_id, limit=None):
+        self.env.cr.execute("""
+            SELECT
+                RATING.rated_partner_id,
+                array_agg(rating) as rating,
+                array_agg(RATING.id) as rating_ids,
+                CASE
+                    WHEN now()::date - RATING.create_date::date BETWEEN 0 AND 6 Then 'week'
+                    WHEN now()::date - RATING.create_date::date BETWEEN 0 AND 15 Then '15 days'
+                    WHEN now()::date - RATING.create_date::date BETWEEN 0 AND 30  Then '1 month'
+                    WHEN now()::date - RATING.create_date::date BETWEEN 0 AND 90  Then '3 month'
+                END AS days
+            FROM
+                rating_rating as RATING
+            LEFT JOIN
+                %s as TASK ON RATING.res_id = TASK.id
+            WHERE
+                RATING.res_model = '%s' AND RATING.rated_partner_id IS NOT NULL AND RATING.create_date >= current_date - interval '90' day AND TASK.project_id = %s AND RATING.rating IN (1,5,10)
+            GROUP BY
+                RATING.rated_partner_id, days, RATING.write_date
+            ORDER BY RATING.write_date desc
+        """ % (table, model, project_id))
```

Comment:
I understand you try to make one query to compute everything (to speed up things). Even, if using the `read_group` from ORM is already good, let's try to make SQL directly. However, the treatment to extract correct stats is very complex, IMHO.
Maybe, we should extract raw data from rating_rating, like 
``` SELECT rated_partner_id, period, sum(R.value) as total, count(R.id) as count, rating_text ``` (Do use average directly, since sum of averages is not equal to average of sum).
where period is the period like you've write before (day_7, day_15, day_30, day_90).

Knowing that, you have 2 stats to compute : periodical and by person.
1) periodical: you know day_30 contains day_15 and day_9 category. With 'count' and 'rating_text' you should be able to compute repartition. After that, calculate percentages.
2) by person: same, group by rated_partner_id the concerned period, and then compute average and repartition.
This should simplify the method.

Also, I've done some naming comment: when reading code, it was less obvious what was handled since `record` variable is not browse record, ...